+nosnoc/+discrete_problem/stewart.m

```diff
@@ -0,0 +1,771 @@
+                        obj.g.eta_const(ii,jj) = {eta_const,
+                            [-inf*ones(n_lambda,1);-inf*ones(n_lambda,1);zeros(n_lambda,1)],
+                            [zeros(n_lambda,1);zeros(n_lambda,1);inf*ones(n_lambda,1)]};
+
+                        obj.g.nu_or(ii,jj) = {[nu-eta;sum(eta)-nu],0,inf};
+
+                        % the actual step eq conditions
+                        %M = 1e5;
+                        M=obj.p.T()/opts.N_stages;
+                        delta_h = obj.w.h(ii,jj) - obj.w.h(ii,jj-1);
+                        step_equilibration = [delta_h + (1/h0)*nu*M;
+                            delta_h - (1/h0)*nu*M];
+                        obj.g.step_equilibration(ii,jj) = {step_equilibration,[0;-inf],[inf;0]};
+                    end
+                end
+            end
+        end
+
+        function populate_problem(obj)
+            obj.create_variables();
+            obj.forward_sim_constraints();
+            obj.generate_complementarities();
+            obj.step_equilibration();
+
+            obj.populated = true;
+        end
+
+        function create_solver(obj, solver_options, plugin)
+            if ~obj.populated
+                obj.populate_problem()
+            end
+
+            if ~exist('plugin')
+                plugin = 'scholtes_ineq';
+            end
+            obj.w.sort_by_index();
+            obj.g.sort_by_index();
+
+            solver_options.assume_lower_bounds = true;
+
+            obj.solver = nosnoc.solver.mpccsol('Mpcc solver', plugin, obj.to_casadi_struct(), solver_options);
+        end
+
+        function stats = solve(obj)
+            opts = obj.opts;
+            T_val = obj.p.T().val;
+
+            for ii=1:opts.N_stages
+                for jj=1:opts.N_finite_elements(ii)
+                    % Recalculate ubh and lbh based on T_val
+                    h0 = T_val/(opts.N_stages*opts.N_finite_elements(ii));
+                    ubh = (1 + opts.gamma_h) * h0;
+                    lbh = (1 - opts.gamma_h) * h0;
+                    if opts.time_rescaling && ~opts.use_speed_of_time_variables
+                        % if only time_rescaling is true, speed of time and step size all lumped together, e.g., \hat{h}_{k,i} = s_n * h_{k,i}, hence the bounds need to be extended.
+                        ubh = (1+opts.gamma_h)*h0*opts.s_sot_max;
+                        lbh = (1-opts.gamma_h)*h0/opts.s_sot_min;
+                    end
+                    obj.w.h(ii,jj).lb = lbh;
+                    obj.w.h(ii,jj).ub = ubh;
+                end
+            end
+
+            stats = solve@vdx.problems.Mpcc(obj);
+        end
+
+        function results = get_results_struct(obj)
+            opts = obj.opts;
+            model = obj.model;
+
+            rbp = ~opts.right_boundary_point_explicit;
+            
+            if opts.right_boundary_point_explicit
+                results.x = obj.discrete_problem.w.x(:,:,obj.opts.n_s).res;
```

Comment:
discrete  -> discrete_time to be more specific

Reply:
Agreed, renamed.